Improved  BlameControlTests

RussKie opened the pull request.

pmiossec commented on UnitTests/GitUITests/UserControls/BlameControlTests.cs:
```diff
@@ -5,13 +5,14 @@
 
 namespace GitUITests.UserControls
 {
+    [SetCulture("en-US")]
```
I like that (a lot more than the deleted lines just below)!

But I'm really not fond of the "en-US" culture chosen (rant: I really dislike their dumb and only used by them date format) but I'm not the center of the universe 😉.

pmiossec commented on UnitTests/GitUITests/UserControls/BlameControlTests.cs:
```diff
@@ -22,25 +23,15 @@ public class BlameControlTests
-            });
-
-            var blameControl = blameControlTask.Join();
-            _sut = new BlameControl.TestAccessor(blameControl);
```
👍

pmiossec commented on UnitTests/GitUITests/UserControls/BlameControlTests.cs:
```diff
@@ -68,68 +61,85 @@ public void SetUp()
+            }
+            finally
+            {
+                AppSettings.BlameShowAuthorTime = originalValue;
```
restoring initial value 👍

RussKie commented:
Thank you!
I saw it failed but didn't have time to fix it.

I don't like en-US locale too, I much prefer en-AU or en-GB myself :)
I changed to en-US because en-GB was failing for me for some strange
reason... It was late and I was tired to debug it.
In all other tests we use en-US too.




On Fri, 10 May 2019 at 22:36, Philippe Miossec <notifications@github.com>
wrote:

> *@pmiossec* approved this pull request.
>
> I have fixed the build.
> But have a look at the commit I pushed (that revert *a little* some
> changes of the PR).
> Feel free to remove my commit and fix the build by introducing the use of
> a local variable.
> ------------------------------
>
> In UnitTests/GitUITests/UserControls/BlameControlTests.cs
> <https://github.com/gitextensions/gitextensions/pull/6544#discussion_r282861718>
> :
>
> >  using GitUI.Blame;
>
>  using GitUIPluginInterfaces;
>
>  using NUnit.Framework;
>
>
>
>  namespace GitUITests.UserControls
>
>  {
>
> +    [SetCulture("en-US")]
>
>
> I like that (a lot more than the deleted lines just below)!
>
> But I'm really not fond of the "en-US" culture chosen (rant: I really
> dislike their dumb and only used by them date format) but I'm not the
> center of the universe 😉.
> ------------------------------
>
> In UnitTests/GitUITests/UserControls/BlameControlTests.cs
> <https://github.com/gitextensions/gitextensions/pull/6544#discussion_r282861989>
> :
>
> > @@ -22,25 +23,15 @@ public class BlameControlTests
>
>          [SetUp]
>
>          public void SetUp()
>
>          {
>
> -            CultureInfo.CurrentCulture = new CultureInfo("EN-GB");
>
> -            var blameControlTask = ThreadHelper.JoinableTaskContext.Factory.RunAsync(async () =>
>
> -            {
>
> -                await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
>
> -                return new BlameControl();
>
> -            });
>
> -
>
> -            var blameControl = blameControlTask.Join();
>
> -            _sut = new BlameControl.TestAccessor(blameControl);
>
>
> 👍
> ------------------------------
>
> In UnitTests/GitUITests/UserControls/BlameControlTests.cs
> <https://github.com/gitextensions/gitextensions/pull/6544#discussion_r282862599>
> :
>
> > +
>
> +                var (gutter, content) = _sut.BuildBlameContents("fileName.txt");
>
> +
>
> +                content.Should().Be($"line1{Environment.NewLine}line2{Environment.NewLine}line3{Environment.NewLine}line4{Environment.NewLine}");
>
> +                var gutterLines = gutter.Split(new[] { Environment.NewLine }, StringSplitOptions.RemoveEmptyEntries);
>
> +                gutterLines[0].TrimEnd().Should().Be("3/22/2010 12:01 PM - author1");
>
> +                gutterLines[1].Should().NotBeNullOrEmpty().And.BeNullOrWhiteSpace();
>
> +                gutterLines[2].TrimEnd().Should().Be("3/22/2011 12:01 PM - author2");
>
> +                gutterLines[3].Should().NotBeNullOrEmpty().And.BeNullOrWhiteSpace();
>
> +
>
> +                // All the lines have the same length
>
> +                gutterLines.Select(l => l.Length).Should().AllBeEquivalentTo(gutterLines[0].Length);
>
> +            }
>
> +            finally
>
> +            {
>
> +                AppSettings.BlameShowAuthorTime = originalValue;
>
>
> restoring initial value 👍
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gitextensions/gitextensions/pull/6544#pullrequestreview-236075615>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABBTEXR3ZJMGBVHRQLCPKGDPUVT3JANCNFSM4HL2DDMQ>
> .
>

pmiossec commented:
>I don't like en-US locale too, I much prefer en-AU or en-GB myself :) 

&

> In all other tests we use en-US too.

I supposed it was the reason but some rant make you sometimes feel better 🤣

RussKie commented:
😃

On Fri, May 10, 2019, 11:47 PM Philippe Miossec <notifications@github.com>
wrote:

> I don't like en-US locale too, I much prefer en-AU or en-GB myself :)
>
> &
>
> In all other tests we use en-US too.
>
> I supposed it was the reason but some rant make you sometimes feel better
> 🤣
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly, view it on GitHub
> <https://github.com/gitextensions/gitextensions/pull/6544#issuecomment-491294801>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABBTEXWLQ7AZDAKPGBDNS6DPUV4IFANCNFSM4HL2DDMQ>
> .
>
